packages/react-compose/src/index.ts

```diff
@@ -13,7 +18,7 @@ export default function compose<Props>(
 ) {
   return function wrapComponent<ComposedProps, C>(
     OriginalComponent: ReactComponent<ComposedProps> & C,
```

Comment:
I think we can maybe do `C extends ReactComponent<ComposedProps>` for the generic, then ditch the `& C` bits here

Reply:
I tried and looks like it doesn't work unfortunately.